### ISSUE TITLE
Select vault not working as intended: not all vaults showing up

### ISSUE THREAD

**Issue by Tycholiz** [dendronhq/dendron#2445]
## Please select the area your bug applies to. (Multiple selections are Possible. You can leave blank if you're not sure.)

- [ ] Workspace.  VSCode workspace, vaults, Intellisense/autocomplete, Dendron settings
- [x] Lookup.  Dendron's Lookup Command
- [ ] Views.  Dendron Preview, Tree View, Side Panels in the UI
- [ ] Schema.  Dendron Schemas
- [ ] Pod.  Data import from / export to Dendron
- [ ] Publish.  External Site Publish
- [ ] Markdown.  Markdown features, such as syntax support and features

## Describe the bug

I am trying to create a new note `aws.sdk`. I have 3 vaults: `personal`, `tech`, `thoughts-on`. I already have an `aws` hierarchy in my `tech` vault. When I try to create the new note, only `personal` and `thoughts-on` appear. When I change the "select vault" option from "smart" to "always prompt", all 3 vaults show up, but when I select `tech`, it prompts me to select again, with only the first 2 vaults shown (see gif to see this behaviour).

![dendron-bug](https://user-images.githubusercontent.com/39745457/154536460-7b7673f0-6ede-4024-8c50-e9a891159a52.gif)

## Desktop (please complete the following information)

- OS: MacOS Monterey 12.2.1
- Browser: Chrome
- Version: 98.0.4758.80

## Dendron Log file

```
{"level":30,"time":1645118366085,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1645118366086,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":8}
{"level":30,"time":1645118366087,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"aws.sd","profile":10,"cancelled":false}
{"level":30,"time":1645118366249,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"aws.sdk","justActivated":false,"prevQuickpickValue":"aws.sd"}
{"level":30,"time":1645118366259,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"post:queryNotes"}
{"level":30,"time":1645118366259,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"createPickerItemsFromEngine","msg":"engine.query","profile":9}
{"level":30,"time":1645118366260,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"aws.sdk","profile":10,"cancelled":false}
{"level":30,"time":1645118368998,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupProvider:onDidAccept","quickpick":"aws.sdk"}
{"level":30,"time":1645118369004,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"getVaultForOpenEditor","msg":"no active doc"}
{"level":30,"time":1645118369004,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"getVaultForOpenEditor","msg":"exit","vault":{"fsPath":"tech","name":"tech"}}
{"level":30,"time":1645118371570,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"dendron.lookupNote:run","msg":"pre-execute"}
{"level":30,"time":1645118371571,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupCommand:execute","msg":"enter"}
{"level":30,"time":1645118374176,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"LookupControllerV3:onHide","msg":"exit"}
{"level":30,"time":1645118374177,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupCommand:execute","msg":"exit"}
{"level":30,"time":1645118374177,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"dendron.lookupNote:run","msg":"post-execute"}
{"level":30,"time":1645118422734,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupCommand:gatherInput","msg":"enter"}
{"level":30,"time":1645118422736,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"LookupControllerV3:new","msg":"enter"}
{"level":30,"time":1645118422737,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"prepareQuickPick","msg":"enter"}
{"level":30,"time":1645118422738,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"prepareQuickPick","msg":"exit"}
{"level":30,"time":1645118422739,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"enter"}
{"level":30,"time":1645118422739,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"onUpdatePickerItems:pre"}
{"level":30,"time":1645118422739,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"","justActivated":true}
{"level":30,"time":1645118422742,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"","profile":2,"cancelled":false}
{"level":30,"time":1645118422742,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"onUpdatePickerItems:post"}
{"level":30,"time":1645118422742,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupProvider.provide","msg":"enter"}
{"level":30,"time":1645118422743,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupProvider.provide","msg":"exit"}
{"level":30,"time":1645118422743,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"exit"}
{"level":30,"time":1645118428029,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"LookupControllerV3:onHide","msg":"exit"}
{"level":30,"time":1645118546356,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupCommand:gatherInput","msg":"enter"}
{"level":30,"time":1645118546357,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"LookupControllerV3:new","msg":"enter"}
{"level":30,"time":1645118546357,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"prepareQuickPick","msg":"enter"}
{"level":30,"time":1645118546358,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"prepareQuickPick","msg":"exit"}
{"level":30,"time":1645118546358,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"enter"}
{"level":30,"time":1645118546358,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"onUpdatePickerItems:pre"}
{"level":30,"time":1645118546358,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"enter","queryOrig":"","justActivated":true}
{"level":30,"time":1645118546359,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"updatePickerItems","msg":"exit","queryOrig":"","profile":1,"cancelled":false}
{"level":30,"time":1645118546360,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"onUpdatePickerItems:post"}
{"level":30,"time":1645118546360,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupProvider.provide","msg":"enter"}
{"level":30,"time":1645118546360,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"NoteLookupProvider.provide","msg":"exit"}
{"level":30,"time":1645118546360,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"showQuickPick","msg":"exit"}
{"level":30,"time":1645118550984,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"LookupControllerV3:onHide","msg":"exit"}
{"level":30,"time":1645118689000,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"dendron.dev.openLogs:run","msg":"pre-execute"}
{"level":30,"time":1645118689003,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"execute","opts":{}}
{"level":30,"time":1645118689119,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"dendron.dev.openLogs:run","msg":"post-execute"}
{"level":30,"time":1645118689385,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"WindowWatcher:onDidChangeVisibleTextEditors","editorPaths":["/Users/kyletycholiz/Library/Application Support/VSCodium/logs/20220215T103910/exthost1/dendron.dendron/dendron.log"]}
{"level":30,"time":1645118689396,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"WorkspaceWatcher:onFirstOpen","msg":"enter","fname":"dendron.log"}
{"level":30,"time":1645118727061,"pid":5832,"hostname":"RIV-907674M","name":"dendron","ctx":"WorkspaceWatcher:onFirstOpen","msg":"exit","fname":"dendron.log"}
```

**Comment by Tycholiz** [comment]
I realized that I already had a note called `aws.SDK`. it wasn't showing up because I looked up with lowercase. I think this is what is causing the issue.

**Comment by SeriousBug** [comment]
Related to #1654

**Comment by Harshita-mindfire** [comment]
Hello @Tycholiz, A fix for #1654 was added in v0.87 of Dendron. It should also fix this bug. Can you confirm if it is still an issue for you?
